packages/application/src/services.tsx

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2018-2020 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Dialog, showDialog } from '@jupyterlab/apputils';
+import * as React from 'react';
+
+import { RequestHandler } from './requests';
+
+/**
+ * A utilities class for various elyra services.
+ */
+export class FrontendServices {
+  static noMetadataError(metadataName: string): Promise<Dialog.IResult<any>> {
+    return showDialog({
+      title: 'Error retrieving metadata',
+      body: <p>No {metadataName} metadata has been configured.</p>,
+      buttons: [Dialog.okButton()]
+    });
+  }
+
+  static async getMetadata(namespace: string): Promise<any> {
+    const metadataResponse: any = await RequestHandler.makeGetRequest(
+      'api/metadata/' + namespace,
+      false
+    );
+
+    if (Object.keys(metadataResponse[namespace]).length === 0) {
+      return this.noMetadataError(namespace);
```

Comment:
i don't think an empty list necessarily means an error retrieving metadata. for example, i could have a `code-snippets` namespace but just not have any code snippets at the moment. in this case an empty list is returned which i think should be valid and not considered an error. i think the error dialog should be presented only if the request returns an actual error.

in addition the use of a dialog here can become tedious. if i tend to leave the code-snippet panel open by default, each time i refresh or relaunch lab i am prompted with the dialog. instead of opening an error dialog return the empty list and leave it up to the caller to decide how to handle the empty list. for example, code snippets could just display some placeholder text (i.e., 'No code snippets have been configured') in the panel when there are no snippets.

Reply:
This is a good argument for having a separate `getRuntimes` function like @kevin-bates and I discussed in the other thread above. Then we could take this check out of the getMetadata call and move it into the specific namespace wrappers that require it (runtimes/runtimeImages). I'll make the changes